cosign/verifier.go

```diff
@@ -0,0 +1,549 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cosign
+
+import (
+	"context"
+	"crypto"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/notaryproject/ratify-go"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
+	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
+	protorekor "github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v1"
+	"github.com/sigstore/sigstore-go/pkg/bundle"
+	"github.com/sigstore/sigstore-go/pkg/root"
+	"github.com/sigstore/sigstore-go/pkg/tuf"
+	"github.com/sigstore/sigstore-go/pkg/verify"
+	"github.com/sigstore/sigstore/pkg/signature"
+)
+
+const (
+	verifierTypeCosign        = "cosign"
+	signatureArtifactCosign   = "application/vnd.dev.cosign.artifact.sig.v1+json"
+	sigstoreBundleMediaType01 = "application/vnd.dev.sigstore.bundle+json;version=0.1"
+	annotationKeyBundle       = "dev.sigstore.cosign/bundle"
+	AnnotationKeyCert         = "dev.sigstore.cosign/certificate"
+	annotationKeySignature    = "dev.cosignproject.cosign/signature"
+	mediaTypeSimpleSigning    = "application/vnd.dev.cosign.simplesigning.v1+json"
+)
```

Comment:
Let's make them consistent and clear.

```suggestion
const (
	verifierTypeCosign = "cosign"
	
	mediaTypeCosignArtifactSignature = "application/vnd.dev.cosign.artifact.sig.v1+json"
	mediaTypeSigstoreBundle01        = "application/vnd.dev.sigstore.bundle+json;version=0.1"
	mediaTypeCosignSimpleSigning     = "application/vnd.dev.cosign.simplesigning.v1+json"

	annotationKeyBundle    = "dev.sigstore.cosign/bundle"
	annotationKeyCert      = "dev.sigstore.cosign/certificate"
	annotationKeySignature = "dev.cosignproject.cosign/signature"
)
```